### DISCUSSION TITLE
align at decimal point

### DISCUSSION THREAD

**Discussion by nschloe** [Textualize/rich#2234] [Ideas]
I need to display numerical data once in a while, e.g.,
```python
from rich.table import Table
from rich import print

table = Table(show_header=False)
table.add_column()
table.add_row("0.567")
table.add_row("-0.1")
table.add_row("+123.4")
table.add_row("+1")
table.add_row("-1")

print(table)
```
```
┏━━━━━━━━┓
┃ 0.567  ┃
│ -0.1   │
│ +123.4 │
│ +1     │
│ -1     │
└────────┘
```
I'd love Rich to be able to align the values at the decimal point (or perhaps more generally: at a given character) to get
```
┏━━━━━━━━━━┓
┃    0.567 ┃
│   -0.1   │
│ +123.4   │
│   +1     │
│   -1     │
└──────────┘
```

**Comment by willmcgugan** [comment]
I'm not sure this is within the remit of the table class, which doesn't know anything about the data other than how to render it. But you could do this with a little pre-processing.
